src/function/memo.rs

```diff
@@ -119,23 +134,42 @@ pub(super) struct Memo<V> {
     /// as the current revision.
     pub(super) verified_at: AtomicCell<Revision>,
 
+    /// Is this memo verified to not be a provisional cycle result?
+    pub(super) verified_final: AtomicCell<bool>,
```

Comment:
Nit: 
```suggestion
    pub(super) verified_final: AtomicBool,
```

Reply:
This change forced me to think about ordering. I'm pretty sure Relaxed is actually fine here, because `verified_final` for a given memo is either always true, or starts false and later becomes true. The latter case is purely an optimization; reading an out-of-date false value just means we might re-validate it unnecessarily; there are no other reads whose validity requires that we get an up-to-date value. I tried to capture this in some added comments; let me know if it's clear enough (or even more so if you think my conclusion is wrong!)